listenbrainz/metadata_cache/apple/client.py

```diff
@@ -41,3 +41,16 @@ def get(self, url, params=None):
                     tokens = AppleService().fetch_access_token()
                     self.developer_token = tokens["access_token"]
             response.raise_for_status()
+
+    def get_user_data(self, url, music_user_token, params=None):
+        with self._get_requests_session() as http:
```

Comment:
This likely won't work because if a music user token expires it cannot be refreshed without manual user intervention. If the developer token expires, the music user token expires too. So I think you should just try once and if there is a 403 error return that from the playlist import endpoint and show a relevant error message on the frontend asking users to reconnect their account.